llvm/lib/MC/MCObjectStreamer.cpp

```diff
@@ -398,10 +392,9 @@ void MCObjectStreamer::emitInstToData(const MCInst &Inst,
   getAssembler().getEmitter().encodeInstruction(Inst, Code, Fixups, STI);
 
   auto CodeOffset = DF->getContents().size();
-  for (MCFixup &Fixup : Fixups) {
+  for (MCFixup &Fixup : Fixups)
     Fixup.setOffset(Fixup.getOffset() + CodeOffset);
-    DF->addFixup(Fixup);
-  }
+  DF->appendFixups(Fixups);
```

Comment:
Check if Fixups is not empty? Most instructions don't have fixups, so we can avoid an out-of-line call in these cases.

Reply:
Done